connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConnectorTest.java

```diff
@@ -120,43 +101,26 @@ public void testInitializeFailure() {
         workerConnector.doShutdown();
         assertStoppedMetric(workerConnector);
 
-        verifyAll();
+        verify(connector).version();
+        verify(offsetStore).start();
+        verify(connector).initialize(any(SourceConnectorContext.class));
+        verify(listener).onFailure(CONNECTOR, exception);
+        verify(listener).onShutdown(CONNECTOR);
+        verify(ctx).close();
+        verify(offsetStorageReader).close();
+        verify(offsetStore).stop();
     }
 
     @Test
     public void testFailureIsFinalState() {
         RuntimeException exception = new RuntimeException();
         connector = sinkConnector;
 
-        connector.version();
-        expectLastCall().andReturn(VERSION);
-
-        connector.initialize(EasyMock.notNull(SinkConnectorContext.class));
-        expectLastCall().andThrow(exception);
-
-        listener.onFailure(CONNECTOR, exception);
-        expectLastCall();
-
-        // expect no call to onStartup() after failure
-
-        listener.onShutdown(CONNECTOR);
-        expectLastCall();
-
-        ctx.close();
-        expectLastCall();
-
-        offsetStorageReader.close();
-        expectLastCall();
-
-        offsetStore.stop();
-        expectLastCall();
-
-        Callback<TargetState> onStateChange = createStrictMock(Callback.class);
-        onStateChange.onCompletion(EasyMock.anyObject(Exception.class), EasyMock.isNull());
-        expectLastCall();
-
-        replayAll();
+        when(connector.version()).thenReturn(VERSION);
+        doThrow(exception).when(connector).initialize(any());
 
+        @SuppressWarnings("unchecked")
+        Callback<TargetState> onStateChange = mock(Callback.class);
```

Comment:
What do you think about introducing a utility method so that creating callbacks like these can be done without having to add the `@SuppressWarnings` annotation?

I'm thinking something like this:
```java
@SuppressWarnings("unchecked")
private Callback<TargetState> mockCallback() {
    return mock(Callback.class);
}
```